fem/nonlininteg.cpp

```diff
@@ -813,4 +816,121 @@ void VectorConvectionNLFIntegrator::AssembleElementGrad(
    }
 }
 
+
+void ConvectiveVectorConvectionNLFIntegrator::AssembleElementGrad(
+   const FiniteElement &el,
+   ElementTransformation &trans,
+   const Vector &elfun,
+   DenseMatrix &elmat)
+{
+   int nd = el.GetDof();
+   int dim = el.GetDim();
+
+   shape.SetSize(nd);
+   dshape.SetSize(nd, dim);
+   dshapex.SetSize(nd, dim);
+   elmat.SetSize(nd * dim);
+   elmat_comp.SetSize(nd);
+   gradEF.SetSize(dim);
+
+   EF.UseExternalData(elfun.GetData(), nd, dim);
+
+   double w;
```

Comment:
I would have this declared within the loop. You could even make use of a ternary operation internally to make it a const double as well by doing something like:

```c++
const double w = Q ? Q->Eval(trans, ip) * ip.weight : ip.weight;
```

Although, I would say that sacrifices readability of the code a bit.

Reply:
The implementation is based on VectorConvectionNLFIntegrator::AssembleElementGrad, which I did not write. I'm open to making this change, however I think it should be consistent with VectorConvectionNLFIntegrator::AssembleElementGrad, which I think is beyond the scope of this PR (even if it's a relatively minor change).

Reply:
@nnpp-schneier so you could definitely make the changes within the classes you've created in this PR. I don't think moving a variable from outside a loop into would make it any harder make to compare the child class against the parent class. 

The ternary operation we should probably get @YohannDudouit and @jandrej thoughts on as the other reviewers. I mean I like them just because they can allow us to make a variable const, but I know not everyone is always as big of a fan of them.

Reply:
Since no one seemed to have an issue with it I went ahead and just made this suggested change.